keps/sig-api-machinery/3352-aggregated-discovery/README.md

```diff
@@ -0,0 +1,788 @@
+updated every time this enhancement is being considered for a
+milestone. -->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!-- This section is incredibly important for producing high-quality,
+user-focused documentation such as release notes or a development
+roadmap. It should be possible to collect this information before
+implementation begins, in order to avoid requiring implementors to
+split their attention between writing release notes and implementing
+the feature itself. KEP editors and SIG Docs should help to ensure
+that the tone and content of the `Summary` section is useful for a
+wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the
+[documentation style guide]. In particular, wrap lines to a reasonable
+length, to make it easier for reviewers to cite specific portions, and
+to minimize diff churn on updates.
+-->
+
+
+The operations that a Kubernetes API server support is dictated
+through a collection of small documents partitioned by group-version.
+All clients of Kubernetes APIs must send a request to every
+group-version in order to "discover" the available APIs. This causes a
+storm of requests for clusters and is a source of latency and
+throttling. This KEP proposes centralizing the "discovery" mechanism
+into one document so that clients only need to make one request to the
+API server to retrieve all the operations available.
+
+## Motivation
+
+<!-- This section is for explicitly listing the motivation, goals, and
+non-goals of this KEP. Describe why the change is important and the
+benefits to users. The motivation section can optionally provide links
+to [experience reports] to demonstrate the interest in a KEP within
+the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports -->
+
+All clients and users of Kubernetes APIs usually first need to
+“discover” what the available APIs are and how they can be used. These
+APIs are described through a mechanism called “Discovery” which is
+typically queried to then build the requests to correct APIs.
+Unfortunately, the “Discovery” API is made of lots of small objects
+that need to be queried individually, causing possibly a lot of delay
+due to the latency of each individual request (up to 80 requests, with
+most objects being less than 1,024 bytes). The more complex the APIs
+provided by the Kubernetes cluster, the more requests need to be
+performed.
+
+The most well known Kubernetes client that uses the discovery
+mechanism is `kubectl`, and more specifically the
+`CachedDiscoveryClient` in `client-go`. To mitigate some of this
+latency, kubectl has implemented a (10 minute timer) 6 hours(!) timer
```

Comment:
10 minutes or 6 hours?

Reply:
It was recently bumped from 10 mins to 6 hours in https://github.com/kubernetes/kubernetes/pull/107141.

Reply:
Copying strikethrough text from google docs to markdown clearly shouldn't be attempted. Should say 6 hours.